bd-client-common/src/safe_file_cache.rs

```diff
@@ -125,28 +152,48 @@ impl<T: Message> SafeFileCache<T> {
     else {
       return Ok((true, None));
     };
+    log::debug!("loaded retry count {retry_count} for {}", self.name);
 
+    // Same for the cached nonce file.
+    let Ok(nonce) =
+      async { read_checksummed_data(&tokio::fs::read(&self.last_nonce_file()).await?) }.await
+    else {
+      return Ok((true, None));
+    };
     log::debug!(
-      "loaded file at retry count {retry_count} for {:?}",
+      "loaded nonce {} for {}",
+      std::str::from_utf8(&nonce).unwrap_or_default(),
       self.name
     );
 
-    // Update the retry count before we apply the config. If this fails, we bail out (which will
-    // attempt to clear the cache directory) and disable caching. We do this because being unable
-    // to update the retry count may result in us getting stuck processing what we think is retry
-    // 0 over and over again.
-    self.persist_cache_load_retry_count(retry_count + 1).await?;
+    {
+      let mut locked = self.locked_state.lock();
+      locked.current_retry_count = retry_count;
+      locked.cached_nonce = nonce;
+    }
 
     // TODO(snowp): Should we read this from runtime as well? It would make it possible for a bad
     // runtime config to accidentally set this really high, but right now this is not
     // configurable at all.
-    if retry_count > MAX_RETRY_COUNT {
-      // Note that eventually if the client is killed this is going to kick in and delete cached
-      // config. This is OK since we are still covered by the kill file, and ultimately we would
-      // like the client to come up and get fresh config anyway.
-      return Ok((true, None));
+    if retry_count >= MAX_RETRY_COUNT {
+      // Note that eventually if the client is killed this is going to kick in since we attempt
+      // to load cached runtime very early on. Since we don't clean state so that we can do changed
+      // nonce detection below, this could pause killed clients to not get new config when they
```

Comment:
What does "below" refer to here? We're just returning in this function

Reply:
Will clarify